packages/studio/src/components/AddPageButton/BasicPageDataCollector.tsx

```diff
@@ -41,6 +42,11 @@ export default function BasicPageDataCollector({
         const getPathValue = data.url
           ? createGetPathVal(data.url, isEntityPage)
           : undefined;
+        const pageDataValidator = new PageDataValidator();
```

Comment:
Let's move the creation of `pageDataValidator` outside of the `onConfirm` callback. Each time `onConfirm` fires, we're creating a new instance of this class which is a bit inefficient. We can just instantiate it once, on creation of `BasicPageDataCollector`.